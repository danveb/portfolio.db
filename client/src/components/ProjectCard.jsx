import "../styles/ProjectCard.css"; 

export default function ProjectCard({ img, alt, title, description }) {
  return (
    <div className="projectCard__card" data-testid="projectCard">
      <img src={img} alt={alt} />
      <h2>{title}</h2>
      <p>{description}</p>
    </div>
  )
}