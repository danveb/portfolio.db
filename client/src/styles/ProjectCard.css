.projectCard__card {
  display: grid; 
  grid-template-columns: repeat(1, 1fr); 
  grid-auto-rows: auto; 
  grid-gap: 2rem; 
  background-color: #fff; 
  padding: 15px; 
}

.projectCard__card img {
  height: 100%; 
  width: 100%; 
  object-fit: cover; 
}

.projectCard__card h3 {
  font-family: "Mulish", sans-serif; 
  font-size: 20px; 
  color: var(--rich-black); 
  letter-spacing: 0.5px; 
}

.projectCard__card p {
  font-family: inherit; 
  font-size: 16px; 
  font-weight: 300; 
  color: var(--raisin-black); 
  letter-spacing: 1px; 
  line-height: 1.5rem; 
}