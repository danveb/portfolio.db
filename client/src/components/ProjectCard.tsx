import "../styles/ProjectCard.css"; 

import { ProjectCardProps } from "../types";

export default function ProjectCard({ img, alt, title, description }: ProjectCardProps) {
  return (
    <div className="projectCard__card" data-testid="projectCard">
      <img src={img} alt={alt} />
      <h2>{title}</h2>
      <p>{description}</p>
    </div>
  )
}