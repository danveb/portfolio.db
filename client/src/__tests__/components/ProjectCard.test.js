import { render, screen } from "@testing-library/react"; 
import ProjectCard from "../../components/ProjectCard"; 
import { BrowserRouter } from "react-router-dom";

// sample props to be passed down 
const props = {
  img: "image",
  alt: "project snapshot", 
  title: "Hello World", 
  description: "This is a mock description",
}; 

// jest.mock
jest.mock("../../components/ProjectCard", () => (props) => {
  return <div className="projectCard__card" data-testid="projectCard">
    <img src={props.img} alt={props.alt} />
    <h2>{props.title}</h2>
    <p>{props.description}</p>
  </div>
})

describe("ProjectCard component", () => {
  // smoke test
  test("renders without crashing", () => {
    render(
      <BrowserRouter>
        <ProjectCard />
      </BrowserRouter>
    ); 
  }); 

  // snapshot test
  test("matches snapshot", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <ProjectCard />
      </BrowserRouter>
    ); 
    expect(asFragment()).toMatchSnapshot(); 
  });

  // getByTestId
  test("matches test id", () => {
    render(
      <BrowserRouter>
        <ProjectCard props={props} />
      </BrowserRouter>
    );
    const testId = screen.getByTestId("projectCard"); 
    expect(testId).toBeInTheDocument(); 
  });
});

