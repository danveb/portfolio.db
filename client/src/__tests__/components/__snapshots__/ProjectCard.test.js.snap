// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectCard component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="projectCard__card"
    data-testid="projectCard"
  >
    <img />
    <h2 />
    <p />
  </div>
</DocumentFragment>
`;
